tests/sals/automated_chatflows/test_vdc_dashboard.py

```diff
@@ -4,26 +4,55 @@
 from jumpscale.loader import j
 from jumpscale.sals.vdc.size import VDC_SIZE
 from solutions_automation.vdc import deployer
-from tests.sals.automated_chatflows.chatflows_base import ChatflowsBase
 from tests.sals.vdc.vdc_base import VDCBase
+from tests.sals.automated_chatflows.chatflows_base import ChatflowsBase
+from parameterized import parameterized_class
 
 
+@parameterized_class(("no_deployment"), [("single",), ("double",)])
 @pytest.mark.integration
 class VDCDashboard(VDCBase):
     @classmethod
     def setUpClass(cls):
         super().setUpClass()
+        no_deployment = "single"
         cls.flavor = "platinum"
         cls.kube_config = cls.deploy_vdc()
-        cls.kube_manager = j.sals.kubernetes.Manager(
-            f"{j.sals.fs.home()}/sandbox/cfg/vdc/kube/{cls.vdc.owner_tname}/{cls.vdc.vdc_name}.yaml"
-        )
+
+        if not cls.kube_config:
+            raise RuntimeError("VDC is not deployed")
+
         j.sals.fs.copy_file(
             f"{j.sals.fs.home()}/sandbox/cfg/vdc/kube/{cls.vdc.owner_tname}/{cls.vdc.vdc_name}.yaml",
             j.sals.fs.expanduser("~/.kube/config"),
         )
-        if not cls.kube_config:
-            raise RuntimeError("VDC is not deployed")
+        cls.kube_manager = j.sals.kubernetes.Manager()
+
+        cls.info("Check that resources are available and ready in 5 min maximum")
+        cls.kube_monitor = cls.vdc.get_kubernetes_monitor()
+        has_resource = False
+        resources_expiry = j.data.time.now().timestamp + 300
+        while j.data.time.now().timestamp < resources_expiry and not has_resource:
+            res = cls.kube_monitor.fetch_resource_reservations()
+            for node in res:
+                if node.has_enough_resources(1000, 1024):
+                    cls.info("Cluster resources are ready")
+                    has_resource = True
+                    break
+
+        cls.info("Check that traefik is ready in 5 min maximum")
+        is_traefik_ready = False
+        traefik_expiry = j.data.time.now().timestamp + 300
+        while j.data.time.now().timestamp < traefik_expiry and not is_traefik_ready:
+            rc, system_pods, err = cls.kube_manager._execute(
+                f"kubectl get pods --kubeconfig {cls.kube_manager.config_path} --namespace kube-system | grep traefik"
+            )
+            running = system_pods.count("Running") + system_pods.count("Completed")
+            all = len(system_pods.splitlines())
+            if running == all:
+                cls.info("Traefik is ready")
+                is_traefik_ready = True
```

Comment:
I think no need for this variable, if traefik is running, you can just break

Reply:
Add assert on it will be better.